src/components/views/TitleForm.js

```diff
@@ -0,0 +1,172 @@
+    handleSubmit: PropTypes.func.isRequired,
+    pristine: PropTypes.bool,
+    submitting: PropTypes.bool,
+    values: PropTypes.object,
+  }
+
+  static defaultProps = {
+    initialValues: {},
+  }
+
+  constructor(props) {
+    super(props);
+    this.accordionStatusRef = React.createRef();
+  }
+
+  getSectionProps(id) {
+    const { form, values = {} } = this.props;
+
+    return {
+      form,
+      id,
+      values,
+    };
+  }
+
+  handleSaveKeyCommand = (e) => {
+    const {
+      handleSubmit,
+      pristine,
+      submitting,
+    } = this.props;
+
+    e.preventDefault();
+
+    if (!pristine && !submitting) {
+      handleSubmit();
+    }
+  }
+
+  renderPaneFooter() {
+    const {
+      handlers,
+      handleSubmit,
+      pristine,
+      submitting,
+    } = this.props;
+
+    return (
+      <PaneFooter
+        renderEnd={(
+          <Button
+            buttonStyle="primary mega"
+            disabled={pristine || submitting}
+            id="clickable-update-title"
+            marginBottom0
+            onClick={handleSubmit}
+            type="submit"
+          >
+            <FormattedMessage id="stripes-components.saveAndClose" />
+          </Button>
+        )}
+        renderStart={(
+          <Button
+            buttonStyle="default mega"
+            id="clickable-cancel"
+            marginBottom0
+            onClick={handlers.onClose}
+          >
+            <FormattedMessage id="stripes-components.cancel" />
+          </Button>
+        )}
+      />
+    );
+  }
+
+  renderFirstMenu() {
+    return (
+      <PaneMenu>
+        <FormattedMessage id="ui-agreements.title.closeEdit">
+          {ariaLabel => (
+            <IconButton
+              aria-label={ariaLabel}
+              icon="times"
+              id="close-title-form-button"
+              onClick={this.props.handlers.onClose}
+            />
+          )}
+        </FormattedMessage>
+      </PaneMenu>
+    );
+  }
+
+  shortcuts = [
+    {
+      name: 'save',
+      handler: this.handleSaveKeyCommand,
+    },
+  ];
+
+  render() {
+    const { handlers: { isSuppressFromDiscoveryEnabled }, values: { name } } = this.props;
+
+    return (
+      <HasCommand
+        commands={this.shortcuts}
+        isWithinScope={checkScope}
+        scope={document.body}
+      >
+        <Paneset>
+          <Pane
+            centerContent
+            defaultWidth="100%"
+            firstMenu={this.renderFirstMenu()}
+            footer={this.renderPaneFooter()}
+            id="pane-title-form"
+            paneTitle={<FormattedMessage id="ui-agreements.editResource" values={{ name }} />}
+          >
+            <TitleManager record={name}>
+              <form id="form-title">
+                <TitleCardInfo {...this.getSectionProps('info')} title={this.props.eresource} />
+                <div className={css.separator} />
+                <TitleFormInfo isSuppressFromDiscoveryEnabled={isSuppressFromDiscoveryEnabled} name={name} />
```

Comment:
You could add all these additional props `title`, `isSuppressFromDiscoveryEnabled` and `name` to the `getSectionProps` function above and just have them passed as `...this.getSectionProps()` to the components.